platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceBean.java

```diff
@@ -353,10 +353,12 @@ public void removeApplication(String appName) {
             bundleLocations.add(info.getLocation());
           }
 
-          for (Bundle bundle : context.getBundles()) {
-            for (BundleInfo info : bundles) {
-              if (info.getLocation().equals(bundle.getLocation())) {
-                metatypeInformation.add(metatypeService.getMetaTypeInformation(bundle));
+          if (context != null) {
+            for (Bundle bundle : context.getBundles()) {
+              for (BundleInfo info : bundles) {
+                if (info.getLocation().equals(bundle.getLocation()) && metatypeService != null) {
```

Comment:
✏️  What would be the point in doing these 2 for loops if metatypeService is null? Just move that condition outside the first for loop.